src/controllers/activegate/reconciler/capability/eec_runtime_config.go

```diff
@@ -0,0 +1,79 @@
+package capability
+
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/activegate/reconciler/statefulset"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const extensionsRuntimeProperties = dynatracev1beta1.InternalFlagPrefix + "extensions."
+
+func getExtensionsFlagsFromAnnotations(instance *dynatracev1beta1.DynaKube) map[string]string {
+	extensionsFlags := make(map[string]string)
+	for flag, val := range dynatracev1beta1.GetInternalFlags(instance) {
+		if strings.HasPrefix(flag, extensionsRuntimeProperties) {
+			runtimeProp := strings.TrimPrefix(flag, extensionsRuntimeProperties)
+			extensionsFlags[runtimeProp] = val
+		}
+	}
+	return extensionsFlags
+}
+
+func buildEecRuntimeConfig(instance *dynatracev1beta1.DynaKube) map[string]interface{} {
+	booleanMap := make(map[string]bool)
+	stringMap := make(map[string]string)
+	longMap := make(map[string]int64)
+
+	for runtimeProp, val := range getExtensionsFlagsFromAnnotations(instance) {
+		if parsedLongInt, err := strconv.ParseInt(val, 10, 64); err == nil {
+			longMap[runtimeProp] = parsedLongInt
+		} else if parsedBool, err := strconv.ParseBool(val); err == nil {
+			booleanMap[runtimeProp] = parsedBool
+		} else {
+			stringMap[runtimeProp] = val
+		}
+	}
+
+	return map[string]interface{}{
```

Comment:
Couldn't this `map[string]interface{}` be an actual type ?
Something like:
```go
type EecRuntimeConfig struct {
	revision	int	`json:"revision"`
	booleanMap	map[string]bool	`json:"booleanMap"`
	stringMap	map[string]string	`json:"stringMap"`
	longMap	map[string]int64	`json:"longMap"`
}
```

Reply:
Sounds great. Why didn't I think about it earlier?

Done 👍